### ISSUE TITLE
Task Page gets ajax error

### ISSUE THREAD

**Issue by pseiler** [openSUSE/open-build-service#10951]
# Issue Description
When I visit the Task page to look up my open tasks in our private instance, I get the followowing errors:
```
DataTables warning: table id=reviews_in_table - Ajax error. For more information about this error, please see http://datatables.net/tn/7
```
If I click okay I get more of these referring to other tables:
```
DataTables warning: table id=requests_in_table - Ajax error. For more information about this error, please see http://datatables.net/tn/7
``` 
```
DataTables warning: table id=requests_out_table - Ajax error. For more information about this error, please see http://datatables.net/tn/7
```
```
DataTables warning: table id=requests_declined_table - Ajax error. For more information about this error, please see http://datatables.net/tn/7
```

```
DataTables warning: table id=all_requests_table - Ajax error. For more information about this error, please see http://datatables.net/tn/7
```
Screenshot of the first error:
![ajax_obs_error](https://user-images.githubusercontent.com/1420501/112850690-88451180-90aa-11eb-8af1-a9320c15d4f3.png)

Also the relevant content is not loading properly. I can see 1 open task in the indicator, but after accepting all the errors I get, the tables showing that there are no tasks.

Expected Result
============
I expect that I get no error and all tasks which should be displayed, are displayed.

How to Reproduce
===============
1. Go to https://example.com/user/tasks
2. See error

Further Information
===============
Operating System: SLES15 SP1 (I know I will update to SP2 asap)
OBS Version: 2.10.8
Installation type (manuall with the documentation).
Package Source: http://download.opensuse.org/repositories/OBS:/Server:/2.10/SLE_15_SP1/
Everything else works great.

I searched the logs and the backend indeed throws a ruby exception.
This is what I see:
```
I, [2021-02-18T13:46:44.557457 #39481]  INFO -- : [3bb60e9e-9c41-4e0b-97b8-c216047eedf3] [39481:461.09] method=GET path=/user/tasks format=html controller=Webui::Users::TasksController action=index status=200 duration=45.73 view=33.76 db=5.91 params={} host=10.129.6.41 time=17105765.30 backend=0 user=<hidden>
F, [2021-02-18T13:46:44.797044 #39481] FATAL -- : [9eba500c-1679-42fa-ae7f-aedca670bef2] [39481:461.33]   
F, [2021-02-18T13:46:44.797134 #39481] FATAL -- : [9eba500c-1679-42fa-ae7f-aedca670bef2] [39481:461.33] ActionController::RoutingError (No route matches [GET] "/users/<hidden>/requests"):
F, [2021-02-18T13:46:44.797207 #39481] FATAL -- : [9eba500c-1679-42fa-ae7f-aedca670bef2] [39481:461.33]   
F, [2021-02-18T13:46:44.797358 #39481] FATAL -- : [9eba500c-1679-42fa-ae7f-aedca670bef2] [39481:461.33] actionpack (5.2.3) lib/action_dispatch/middleware/debug_exceptions.rb:65:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] actionpack (5.2.3) lib/action_dispatch/middleware/show_exceptions.rb:33:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] lograge (0.11.2) lib/lograge/rails_ext/rack/logger.rb:15:in `call_app'
[9eba500c-1679-42fa-ae7f-aedca670bef2] railties (5.2.3) lib/rails/rack/logger.rb:26:in `block in call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] activesupport (5.2.3) lib/active_support/tagged_logging.rb:71:in `block in tagged'
[9eba500c-1679-42fa-ae7f-aedca670bef2] activesupport (5.2.3) lib/active_support/tagged_logging.rb:28:in `tagged'
[9eba500c-1679-42fa-ae7f-aedca670bef2] activesupport (5.2.3) lib/active_support/tagged_logging.rb:71:in `tagged'
[9eba500c-1679-42fa-ae7f-aedca670bef2] railties (5.2.3) lib/rails/rack/logger.rb:26:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] actionpack (5.2.3) lib/action_dispatch/middleware/remote_ip.rb:81:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] request_store (1.4.1) lib/request_store/middleware.rb:19:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] actionpack (5.2.3) lib/action_dispatch/middleware/request_id.rb:27:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/method_override.rb:22:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/runtime.rb:22:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] activesupport (5.2.3) lib/active_support/cache/strategy/local_cache_middleware.rb:29:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] actionpack (5.2.3) lib/action_dispatch/middleware/executor.rb:14:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/lock.rb:17:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/sendfile.rb:111:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] experimental-influxdb-rails (1.0.0.beta5) lib/influxdb/rails/rack.rb:14:in `_call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] experimental-influxdb-rails (1.0.0.beta5) lib/influxdb/rails/rack.rb:9:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] railties (5.2.3) lib/rails/engine.rb:524:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] railties (5.2.3) lib/rails/railtie.rb:190:in `public_send'
[9eba500c-1679-42fa-ae7f-aedca670bef2] railties (5.2.3) lib/rails/railtie.rb:190:in `method_missing'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/urlmap.rb:68:in `block in call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/urlmap.rb:53:in `each'
[9eba500c-1679-42fa-ae7f-aedca670bef2] rack (2.0.7) lib/rack/urlmap.rb:53:in `call'
[9eba500c-1679-42fa-ae7f-aedca670bef2] passenger (6.0.2) src/ruby_supportlib/phusion_passenger/rack/thread_handler_extension.rb:97:in `process_request'
[9eba500c-1679-42fa-ae7f-aedca670bef2] passenger (6.0.2) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:149:in `accept_and_process_next_request'
[9eba500c-1679-42fa-ae7f-aedca670bef2] passenger (6.0.2) src/ruby_supportlib/phusion_passenger/request_handler/thread_handler.rb:110:in `main_loop'
[9eba500c-1679-42fa-ae7f-aedca670bef2] passenger (6.0.2) src/ruby_supportlib/phusion_passenger/request_handler.rb:415:in `block (3 levels) in start_threads'
[9eba500c-1679-42fa-ae7f-aedca670bef2] passenger (6.0.2) src/ruby_supportlib/phusion_passenger/utils.rb:113:in
```

Unfortunately I cannot send you the URL to the instance as it's hosted in the private network of our company.

**Comment by pseiler** [comment]
Is there anything I can do to help you guys indentify the root cause?

**Comment by hennevogel** [comment]
@pseiler nope, this seems pretty clear

**Comment by pseiler** [comment]
Alright. This sounds like good news to me. Looking forward to get this fixed!
If I should test you some patches just mention me here.

**Comment by pseiler** [comment]
@hennevogel Is there any news on this?
Best regards

**Comment by roolebo** [comment]
The missing route was introduced in https://github.com/openSUSE/open-build-service/commit/4f07d4c06c6. The code is still present in 2.10 but for some reason the route definition doesn't work.

```rake routes``` says that the route is available:
```
RAILS_ENV=production bin/rake routes | awk '$3 ~ /^\/users/'
                                          users GET                 /users(.:format)                                                                                                     webui/user#index
                             user_announcements POST                /users/announcements/:id(.:format)                                                                                   webui/users/announcements#create
                                  user_requests GET                 /users(/:user)/requests(.:format)                                                                                    webui/users/bs_requests#index
                                 user_rss_token POST                /users/rss_tokens(.:format)                                                                                          webui/users/rss_tokens#create
```

@hennevogel any ideas?

**Comment by hellcp** [comment]
To me it looks to happen with any username that contains a dot character, maybe that's a pointer as to why it may not be working?

**Comment by pseiler** [comment]
I can confirm that in my case usernames also contains dots. We use ldap usernames and the dots are part of the naming scheme.

**Comment by hellcp** [comment]
This is fairly easy to fix it seems, by default rails paramerers in routes don't accept dots, as per https://guides.rubyonrails.org/routing.html#specifying-constraints, a constraint that accepts dots needs to be defined to work around the issue.

**Comment by dmarcoux** [comment]
Closed by #12044
